pkg/s3/fetch.go

```diff
@@ -0,0 +1,196 @@
+package s3
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type Fetcher struct {
+	Logger *log.Logger
+}
+
+func (f *Fetcher) Fetch(dir string, s3URL string, s3AccessKey string, s3SecretKey string, s3Bucket string, s3File string, s3ForcePathStyle string, s3Region string) error {
+	blob, err := url.Parse(s3URL)
+	if err != nil {
+		return err
+	}
+
+	option := session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}
+
+	forcePath, err := strconv.ParseBool(s3ForcePathStyle)
+	if err != nil {
+		forcePath = false
+	}
+
+	region := s3Region
+	if region == "" {
+		region = "us-west-2"
```

Comment:
should we default this server-side? unclear to me why this specific default.

Reply:
Changed to a stub string. This field is required by amazon-sdk